qiskit_dynamics/solvers/solver_classes.py

```diff
@@ -585,6 +903,12 @@ def setup_simulation_lists(
     elif isinstance(signals, list) and isinstance(signals[0], tuple):
         # multiple lindblad
         multiple_sims = True
+    elif isinstance(signals, (Schedule, ScheduleBlock)):
```

Comment:
Also, type hint only says `Schedule`. Is that ok?

Reply:
What do you think? I figured the type hint for the `signals` argument was already getting kind of crazy, so thought using `Schedule` for both `Schedule` and `ScheduleBlock` was okay? It's definitely not the most formal choice, but also the size of this type hint is growing to the point that it's somewhat unreadable.

I'm happy to either leave it as is, or change it, depending on your thoughts on it.

Reply:
I am assuming that the user specifies the schedules at only one point. Here they can be `Schedule` or `ScheduleBlock` and we explicitly write out the type hint. Then, we convert `ScheduleBlock` to `Schedule` and only ever deal with `Schedule` going forward. It think that's more or less how you have it now?

Reply:
I've just:
- Added `ScheduleBlock` to the `solve` type hint. 
- Changed the `Solver.solve` method to check for `ScheduleBlock`s and convert them to `Schedule`s right at the beginning, and modified all subsequent code to only work with `Schedule`s.

This is good - makes the rest of the code cleaner. I think I had originally been trying to do the conversion as late as possible, but now that I think about it I can't think of any good reason to do that.